server/controllers/profile.js

```diff
@@ -0,0 +1,69 @@
+const mongoose = require('mongoose');
+const Profile = require('../models/Profile');
+const asyncHandler = require('express-async-handler');
+
+// POST /profile - Create new profile
+exports.createNewProfile = asyncHandler(async (req, res) => {
+  const profile = req.body;
+  const newProfile = new Profile({ ...profile });
+
+  if (!newProfile) {
```

Comment:
We won't be handling any server errors here as we are saving to the db after this check is being performed, maybe consider moving this?

Reply:
I'll just remove it for now and if you have any ideas where we want to do error handling please let me know.